modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -25,7 +26,7 @@ package object engine {
     * This represents an actual real-world action to be done in the underlying
     * systems.
     */
-  type Action = Task[Result]
+  type Action = Kleisli[Task, ActionMetadata, Result]
```

Comment:
Right, maybe unconstrained is too *strong* of a word. But it's still not clear to me what criteria to use when deciding if it's worth introducing transformers or not.

One easy rule of thumb I was trying to follow is that it's fine as long as it's only 1 layer, but in this case it's not clear this meets the rule because the `Kleisli` Task will be used in code that will be already under in `Handle` which is monad transformer which is made of `StateT and `Task`, so we'll have some code that hidden under the synonyms will be a StateT/Kleisli/Task monad stack.

Again, this is fine with me, but just wanted to make sure it's worth introducing the `Kleisli` if we can do the same with a simple case class.